File: backend/hotelEase/hotelEase.Model/Reservation.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace hotelEase.Model
{
    public class Reservation
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public int RoomId { get; set; }

        public DateTime CheckInDate { get; set; }

        public DateTime CheckOutDate { get; set; }

        public decimal TotalPrice { get; set; }

        public string? Status { get; set; }

        public DateTime? CreatedAt { get; set; }
        public bool? IsDeleted { get; set; }
        public DateTime? DeletedTime { get; set; }
        public virtual User User { get; set; } = null!;
        public virtual Room? Room { get; set; } = null!;

    }
}

File: UI/hotelease_desktop/lib/models/reservation.dart
import 'package:hotelease_mobile_new/models/payment.dart';
import 'package:hotelease_mobile_new/models/room.dart';
import 'package:hotelease_mobile_new/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int? id;
  int? userId;
  int? roomId;
  DateTime? checkInDate;
  DateTime? checkOutDate;
  double? totalPrice;
  String? status;
  DateTime? createdAt;
  bool? isDeleted;
  DateTime? deletedTime;
  Room? room;

  User? user;
  List<Payment>? payments;

  Reservation({
    this.room,
    this.id,
    this.userId,
    this.roomId,
    this.checkInDate,
    this.checkOutDate,
    this.totalPrice,
    this.status,
    this.createdAt,
    this.isDeleted,
    this.deletedTime,
    this.user,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
